python/union_find.py

```diff
@@ -0,0 +1,94 @@
+from itertools import product
+from typing import Any, Tuple
+
+from mage.union_find import DisjointSet
+
+import mgp
+
+
+disjoint_set = DisjointSet(node_ids=None)
+
+
+class Mode:
+    """
+    Valid values of the mode parameter
+    """
+
+    PAIRWISE = ["p", "pairwise"]
+    CARTESIAN = ["c", "cartesian"]
+
+
+@mgp.read_proc
+def connected(
+    ctx: mgp.ProcCtx,
+    nodes1: Any,
+    nodes2: Any,
+    mode: str = "pairwise",
+    update: bool = True,
+) -> mgp.Record(node1=mgp.Vertex, node2=mgp.Vertex, connected=bool):
+    """
+    Returns whether two nodes (or each pair in the product of two node lists) belong to the same connected component
+    of the graph.
+    :param nodes1: Node or tuple of nodes
+    :type nodes1: Union[mgp.Vertex, Tuple[mgp.Vertex]]
+    :param nodes2: Node or tuple of nodes
+    :type nodes2: Union[mgp.Vertex, Tuple[mgp.Vertex]]
+    :param mode: Mode of operation: `p` / `pairwise` for pairwise similarity calculation or `c` / `cartesian`
+    for calculating the similarity of the Cartesian product of given tuples. Default value is `pairwise`.
```

Comment:
This part of the documentation needs to be fixed.

Reply:
Done!